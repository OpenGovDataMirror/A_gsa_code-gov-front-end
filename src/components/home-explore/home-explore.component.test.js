import React from 'react'
import { shallow } from 'enzyme'
import HomeExplore from 'components/home-explore/home-explore.component'

const props = {
  exploreItems: [
    { title: 't1', description: 'd1', links: [{ name: 'n1', url: 'u1' }] },
    { title: 't2', description: 'd2', links: [{ name: 'n2', url: 'u2' }] },
    { title: 't3', description: 'd3', links: [{ name: 'n3', url: 'u3' }] }
  ]
}

let wrapper

describe('components - HomeExplore', () => {
  beforeEach(() => {
    wrapper = shallow(<HomeExplore {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map all the `exploreItems`', () => {
    expect(wrapper.find('.explore-action').length).toBe(props.exploreItems.length)
  })
})
