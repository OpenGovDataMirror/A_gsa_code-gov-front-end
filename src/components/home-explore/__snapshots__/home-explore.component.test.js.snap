// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - HomeExplore should render correctly 1`] = `
<div
  className="grid-row"
>
  <section
    className="explore usa-section"
    id="explore"
  >
    <div
      className="explore-actions grid-row text-center tablet:border-bottom-2px tablet:border-top-2px tablet:padding-top-205 padding-top-3 tablet:padding-bottom-205 padding-bottom-0"
    >
      <div
        className="explore-action grid-row flex-column flex-justify  padding-right-2 padding-left-2 tablet:grid-col-4 tablet:border-right-2px margin-bottom-5 tablet:margin-bottom-0"
        key="t1"
      >
        <div>
          <h2
            className="text-primary"
          >
            t1
          </h2>
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "d1",
              }
            }
          />
        </div>
        <div
          className="buttons"
        >
          <LinkButtons
            className="usa-button font-body-sm tablet-lg:padding-x-5 line-height-sans-2"
            key="n1"
            name="n1"
            url="u1"
          />
        </div>
      </div>
      <div
        className="explore-action grid-row flex-column flex-justify  padding-right-2 padding-left-2 tablet:grid-col-4 tablet:border-right-2px margin-bottom-5 tablet:margin-bottom-0"
        key="t2"
      >
        <div>
          <h2
            className="text-primary"
          >
            t2
          </h2>
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "d2",
              }
            }
          />
        </div>
        <div
          className="buttons"
        >
          <LinkButtons
            className="usa-button font-body-sm tablet-lg:padding-x-5 line-height-sans-2"
            key="n2"
            name="n2"
            url="u2"
          />
        </div>
      </div>
      <div
        className="explore-action grid-row flex-column flex-justify  padding-right-2 padding-left-2 tablet:grid-col-4 tablet:border-right-2px margin-bottom-5 tablet:margin-bottom-0"
        key="t3"
      >
        <div>
          <h2
            className="text-primary"
          >
            t3
          </h2>
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "d3",
              }
            }
          />
        </div>
        <div
          className="buttons"
        >
          <LinkButtons
            className="usa-button font-body-sm tablet-lg:padding-x-5 line-height-sans-2"
            key="n3"
            name="n3"
            url="u3"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
